src/components/decimal/decimal.component.js

```diff
@@ -11,247 +16,261 @@ import LocaleContext from "../../__internal__/i18n-context";
    */
-  toStandardDecimal = (i18nValue) => {
-    const valueWithoutNBS =
-      this.getSeparator("group").match(/\s+/) && !i18nValue.match(/\s{2,}/)
-        ? i18nValue.replace(/\s+/g, "")
-        : i18nValue;
-    /* If a value is passed in that is a number but has too many delimiters in succession, we want to handle this
+  const toStandardDecimal = useCallback(
+    (i18nValue) => {
+      const valueWithoutNBS =
+        getSeparator("group").match(/\s+/) && !i18nValue.match(/\s{2,}/)
+          ? i18nValue.replace(/\s+/g, "")
+          : i18nValue;
+      /* If a value is passed in that is a number but has too many delimiters in succession, we want to handle this
     value without formatting it or removing delimiters. We also want to consider that,
     if a value consists of only delimiters, we want to treat that 
     value in the same way as if the value was NaN. We want to pass this value to the 
     formatValue function, before the delimiters can be removed. */
-    const errorsWithDelimiter = new RegExp(
-      `([^A-Za-z0-9]{2,})|(^[^A-Za-z0-9-]+)|([^0-9a-z-,.])|([^0-9-,.]+)|([W,.])$`,
-      "g"
-    );
-    const separator = this.getSeparator("decimal");
-    const separatorRegex = new RegExp(
-      separator === "." ? `\\${separator}` : separator,
-      "g"
-    );
-    if (
-      valueWithoutNBS.match(errorsWithDelimiter) ||
-      (valueWithoutNBS.match(separatorRegex) || []).length > 1
-    ) {
-      return valueWithoutNBS;
+      const errorsWithDelimiter = new RegExp(
+        `([^A-Za-z0-9]{2,})|(^[^A-Za-z0-9-]+)|([^0-9a-z-,.])|([^0-9-,.]+)|([W,.])$`,
+        "g"
+      );
+      const separator = getSeparator("decimal");
+      const separatorRegex = new RegExp(
+        separator === "." ? `\\${separator}` : separator,
+        "g"
+      );
+      if (
+        valueWithoutNBS.match(errorsWithDelimiter) ||
+        (valueWithoutNBS.match(separatorRegex) || []).length > 1
+      ) {
+        return valueWithoutNBS;
+      }
+
+      const withoutDelimiters = removeDelimiters(valueWithoutNBS);
+      return withoutDelimiters.replace(new RegExp(`\\${separator}`, "g"), ".");
+    },
+    [getSeparator, removeDelimiters]
+  );
+
+  const decimalValue = getSafeValueProp(defaultValue || value || emptyValue);
+  const [stateValue, setStateValue] = useState(
+    isNaN(toStandardDecimal(decimalValue))
+      ? decimalValue
+      : formatValue(decimalValue)
+  );
+
+  const createEvent = useCallback(
+    (formatted, raw) => {
+      return {
+        target: {
+          name,
+          id,
+          value: {
+            formattedValue: formatValue(toStandardDecimal(formatted)),
+            rawValue: raw || toStandardDecimal(formatted),
+          },
+        },
+      };
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+
+  /**
+   * Determine if the component is controlled at the time of call
+   */
+  function isControlled(arg) {
```

Comment:
I think this should just return `value !== undefined` rather than taking an arg and evaluating it

Reply:
Also if we are going to remove the invariant around switching between controlled/ uncontrolled do we rally need this function at all, could we not just check `value !== undefined` on line 242 instead of ` const isComponentControlled = isControlled(value);`

Reply:
Could be a good shout doing it that way. I'll find out if we need the invariant for controlled/uncontrolled switching and make that changes based on the outcome.